### PR TITLE
HYDRA-801 : Add a test for Usd stages variants

### PR DESCRIPTION
**Pull request by lanierd-adsk**

### CODE REVIEW

**Comment by lanierd-adsk** [comment]
FYI : @lilike-adsk @ppt-adsk  @roopavr-adsk  @vlasovi

File: test/lib/mayaUsd/render/mayaToHydra/testStageVariants.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                "testStageVariants.ma")
+        self.setHdStormRenderer()
+        cmds.refresh()
+
+        shapeNode = "variantsShape"
+        shapeStage = mayaUsdLib.GetPrim(shapeNode).GetStage()
+        
+        #Check geometry variant
+        rootPrim = shapeStage.GetPrimAtPath('/Cubes')
+        self.assertFalse(rootPrim== None)
```
Comment (ppt-adsk) [inline test/lib/mayaUsd/render/mayaToHydra/testStageVariants.py]: Minor: better to use assertIsNotNone() when testing against None, provides clearer output on failure:
https://codereview.doctor/features/python/best-practice/use-assert-is-not-none

File: test/lib/mayaUsd/render/mayaToHydra/testStageVariants.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        self.assertFalse(modVariant== None)
+        self.assertEqual(modVariant.GetVariantSelection(), 'OneCube')
+
+        #Select the USD Cubes to see the selection highlight
+        cubesPath = ufe.Path([
+            mayaUtils.createUfePathSegment("|variants|variantsShape"), 
+            usdUtils.createUfePathSegment("/Cubes")])
+        cubesItems = ufe.Hierarchy.createItem(cubesPath)
+        self.assertIsNotNone(cubesItems)
+        ufe.GlobalSelection.get().clear()
```
Comment (ppt-adsk) [inline test/lib/mayaUsd/render/mayaToHydra/testStageVariants.py]: Minor: awkward that you're always getting the global selection singleton, just assign it to a name:
sn = ufe.GlobalSelection.get()

File: test/lib/mayaUsd/render/mayaToHydra/testStageVariants.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        self.assertTrue(cube2Prim.IsInstanceable())
+        cube3Prim.SetInstanceable(True)
+        self.assertTrue(cube3Prim.IsInstanceable())
+        modVariant.SetVariantSelection('OneCube')
+        self.assertEqual(modVariant.GetVariantSelection(), 'OneCube')
+        ufe.GlobalSelection.get().append(cubesItems)
+        self.assertSnapshotClose("oneCubeInstanceable.png", self.IMAGE_DIFF_FAIL_THRESHOLD, self.IMAGE_DIFF_FAIL_PERCENT)
+        ufe.GlobalSelection.get().clear()
+
+        #Enable the following tests when HYDRA-820 is fixed
```
Comment (ppt-adsk) [inline test/lib/mayaUsd/render/mayaToHydra/testStageVariants.py:107]: Hope you have the task of enabling this test as an acceptance criterion in HYDRA-820 :)

Comment (lanierd-adsk) [reply test/lib/mayaUsd/render/mayaToHydra/testStageVariants.py:107]: Yes, I've added a TODO in the task.

**Review by ppt-adsk — APPROVED** [review]

**Pull request merged**